VLogProcessor/src/edu/umich/robustopt/algorithms/NonConvexDesigner.java

```diff
@@ -594,6 +594,8 @@ private ClusteredWindow createAValidClusteredWindow(Map<Cluster, Double> allClus
 				int freq = (int) Math.round(allClusters.get(c) * normalizingRatio);
 				if (freq >= 1) { // otherwise, we do not want to create empty clusters!
 					Cluster newCluster = workloadGenerator.createClusterWithNewFrequency(c, freq);
+					if (newCluster == null)
+						throw new Exception("allPossibleQueries file has to include all queries in target query_file");
```

Comment:
When allPossibleQuery file doesn't contain all queries in designQuery file, createClusterWithNewFrequency will possibly return null. The reason is that when we change the frequency of a cluster in initial window, we want to add the queries from allPossibleQuery file to that cluster.